### PR TITLE
Extend diff expander (see more lines) button

### LINKED ISSUE #547: Duplicate the expand button in the middle on split diffs
When you are looking at a diff, you have an Expand button to show more context around the changed lines.

When you are using the Unified diff, the expand button is on the left and it's all good.

When you are using the Split diff, though, the expand button is still only on the left and it's pushed too far to the left.

I suggest to either duplicate or move the expand button on Split diffs to be present just in the middle between the left and the right side of the diffs.

### PR DESCRIPTION
**Pull request by donocode**
Allow the entire expandable line to be clicked to see more lines, rather than just the expander button in the left margin. Includes hover style to match the button.

Closes #547

### CODE REVIEW

**Comment by donocode** [comment]
diff-expand

Example of feature in use

**Comment by bfred-it** [comment]
I think this can/should be done with a single delegated event because large hidden diffs can be shown later and also new pushed commits will update the page (in not sure if that triggers the full ajaxHandler function again.

**Comment by donocode** [comment]
Yeah, you're right I'll move the handler up to cover all diffs with one handler.

File: source/content.css [hunk 1]
```diff
@@ -822,15 +822,10 @@ body > .footer li a {
 }
 
 /* For 'extend-diff-expander' feature */
-.refined-github-diff-expand .js-expandable-line:hover .diff-expander,
-.refined-github-diff-expand .js-expandable-line:hover .blob-code {
+.js-expandable-line:hover .diff-expander,
```
Comment (donocode) [inline source/content.css]: This prevents the css from being disabled with the feature

File: source/features/extend-diff-expander.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import delegate from 'delegate';
 function expandDiff(event) {
 	// Skip if the user clicked directly on the icon
 	if (!event.target.closest('.js-expand')) {
-		select('.js-expand', event.target.parentElement).click();
+		select('.js-expand', event.currentTarget).click();
```
Comment (bfred-it) [inline source/features/extend-diff-expander.js]: This doesn't work. `.currentTarget` is the `.diff-view` element so `select` will find the first expander of the page.

File: source/content.css (hunk 1 above)
Comment (bfred-it) [reply source/content.css]: Good point, but I'm not sure of why that would be necessary. If deemed necessary, a single `rgh-diff-expander` class can be added to `.diff-view` so all dynamically-added elements will be handled.

File: source/features/extend-diff-expander.js (hunk 2 above)
Comment (donocode) [reply source/features/extend-diff-expander.js]: I tested it, `delegateTarget` is diff-view and `currentTarget` is js-expandable-line

Comment (bfred-it) [reply source/features/extend-diff-expander.js]: It's not the right one. Try again. `select` picks the first element it finds. Try on a longer page and click the last expander.

Comment (bfred-it) [reply source/features/extend-diff-expander.js]: For example, click the last expander at the bottom of this page: https://github.com/sindresorhus/refined-github/pull/935/files

Comment (donocode) [reply source/features/extend-diff-expander.js]: Yeah you're right, I read the delegate readme incorrectly. `delegateTarget` is actually .js-expandable-line

**Comment by bfred-it** [comment]
This can be merged once it's mentioned in the readme under _UI improvements_

**Comment by bfred-it** [comment]
Thank you @donocode!

**Comment by donocode** [comment]
The image link actually needs to be `.PNG` 😬 good old snipping tool and case sensitive routing

**Comment by borekb** [comment]
For those trying to suppress the (relatively aggressive) blue hover color, this CSS can be used:

```
.js-expandable-line:hover .diff-expander,
.js-expandable-line:hover .blob-code {
	border-color: inherit;
	color: rgba(27,31,35,0.5);
	background-color: #f1f8ff;
}
```

**Comment by agavra** [comment]
Thanks @borekb! This is super helpful, though the code has changed a little bit. For those who are wondering:
```
.js-expandable-line:hover .blob-num:not(:hover) .directional-expander:first-child,
.js-expandable-line:hover .blob-num:not(:hover)+.blob-code {
	border-color: inherit;
	color: rgba(27,31,35,0.5);
	background-color: #f1f8ff;
}
```

**Comment by gaelduplessix** [comment]
The css posted above didn't work for me, here's an updated version:

```css
.js-expandable-line:hover .blob-num:not(:hover) .directional-expander:first-child,
.js-expandable-line:hover .blob-num:not(:hover)+.blob-code {
	border-color: inherit;
	color: var(--color-diff-blob-hunk-text);
	background-color: var(--color-diff-blob-hunk-line-bg);
}
```

@donocode This css change was the first thing I looked up after installing "Refined Github", the hover style is especially surprising in dark mode. It would be nice to allow disabling the effect, at least if the feature is disabled

**Comment by Mordavolt** [comment]
Thanks, folks! I don't know if something has changed, but the above version was not that great for me in dark mode, so I've updated it.
```
.js-expandable-line:hover .blob-num:not(:hover) .directional-expander:first-child,
.js-expandable-line:hover .blob-num:not(:hover)+.blob-code {
    background-color: var(--color-accent-subtle);
    color: var(--color-fg-muted);
}
```

Disabling the feature disables the functionality but not the visuals. Is that something that the maintainers could address?

**Pull request merged**